src/main/java/io/openliberty/tools/common/plugins/util/BinaryScannerUtil.java

```diff
@@ -177,7 +181,7 @@ public Set<String> runBinaryScanner(Set<String> currentFeatureSet, List<String>
                 } else if (scannerException.getClass().getName().equals(FEATURE_CONFLICT_EXCEPTION)) {
                     // The scanned files conflict with each other or with current features
                     Set<String> conflicts = getFeatures(scannerException);
-                    Set<String> sampleFeatureList = reRunBinaryScanner(allClassesDirectories, logLocation, targetJavaEE, targetMicroProfile);
+                    Set<String> sampleFeatureList = reRunIfFailed ? reRunBinaryScanner(allClassesDirectories, logLocation, targetJavaEE, targetMicroProfile): null;
```

Comment:
As Paul mentioned in https://github.com/OpenLiberty/ci.common/pull/362/files#r862254383, the error here (in the case of an "optimize" failure), would always be "A working set of features could not be generated due to conflicts in the application’s API usage", hence the sampleFeatureList will always be `null`. So this line is good as is.